src/Client.re

```diff
@@ -190,17 +190,23 @@ type response('response) =
 type clientResponse('response) = {
   data: option('response),
   error: option(CombinedError.t),
+  extensions: option(Js.Dict.t(string)),
```

Comment:
We were missing `extensions` 😱 Unfortunately the `urql` type for this field, `Record<string, any>` is hard to bind without introducing _lots_ of type parameters. At the same time, `Js.Dict.t(string)` limits extensions to being JS objects with only string values 😞 I'd like to create a better solution down the road.

Reply:
Indeed, sorry, I forgot to report this ^^

An alternative could be an opaque type maybe?

```reasonml
type extension;

extensions: option(Js.Dict.t(extension))
```

If `any` in urql stands for "any _json_ value", then `extensions: option(Js.Json.t)` should work?

Reply:
Well `extensions` in the GraphQL spec are a well-ish defined structured. Per the GraphQL spec:

> The response map may also contain an entry with key extensions. This entry, if set, must have a map as its value.

https://spec.graphql.org/June2018/#sec-Response-Format

So we know `extensions` will always be a map (read: object) structure in JS,  but the types of the values _may_ be polymorphic. In the vast majority of cases I'd expect all values to be `strings`, but this isn't really a safe assumption.

I do like your idea of using `Js.Json.t` and then allowing users to decode it. Their API can set the shape of extensions and they can write their own decoders using the `Js.Json` module 👍